src/c++/perf_analyzer/test_inference_profiler.cc

```diff
@@ -346,4 +373,73 @@ TEST_CASE("testing the GetMeanAndStdDev function")
   }
 }
 
+TEST_CASE("testing the MergeMetrics function")
```

Comment:
Do we need a test for missing gpu metric too? For example, if power usage can't be collected does aggregation work properly?